test/test.baseexternalclient.ts

```diff
@@ -423,29 +424,119 @@ describe('BaseExternalAccountClient', () => {
   });
 
   describe('getProjectId()', () => {
-    it('should return workforcePoolUserProject if client auth is not defined', async () => {
+    it('should resolve with workforce projectID if no client auth not and workforce user project are defined', async () => {
```

Comment:
I think this test is sufficient. You don't need the next one. You already cover the different scenarios above.
Rename to: should resolve for workforce pools when workforce_pool_user_project is provided

Reply:
Can you also add a test in `describe('projectNumber', () => {` to confirm projectNumber is null for workforce pools with workforce_pool_user_project?
```
    it('should return null for workforce pools with workforce_pool_user_project', () => {
      const options = Object.assign(
        {},
        externalAccountOptionsWorkforceUserProject
      );
      const client = new TestExternalAccountClient(options);

      assert(client.projectNumber === null);
    });
```